### DISCUSSION TITLE
TypeScript LSP Experience

### DISCUSSION THREAD

**Discussion by SargisPlusPlus** [LazyVim/LazyVim#3963] [Q&A]
Hello all. 
I have been using LazyVim for about 6 months now. I always keep my plugins up to date.
Recently, (maybe within last 1-2 months after so many plugin updates), I noticed that TypeScript LSP has gotten really slow, to the point of not being usable. It takes ~3 seconds for anything to load up or errors to clear up. It crashes every maybe 10min, I have to run `LspRestart`.

Initially, I enabled typescript-lang through LazyVim Extras. I still have that enabled, but I also now added `lspconfig.lua`

But this did not help.

My specs:
- NVIM v0.10.0
- Mac OS M3
- LazyVim 12.27.0

My lazyvim configs: https://github.com/SargisPlusPlus/nvim

**Comment by yioneko** [comment]
Try to set `typescript.tsserver.maxTsServerMemory = 8192` (or a larger value depends on your project scale). VSCode defaults it to 3072 and that maybe too small for your project to work smoothly.

**Comment by Masihbalali** [comment]
Dealing with performance issues in the TypeScript LSP can be frustrating, especially when it affects your workflow. Here are some steps you can take to diagnose and hopefully resolve the slow performance and crashes:

1. Review Configuration
Double-check your configuration to ensure there are no conflicting settings or plugins that might be affecting the performance of the TypeScript LSP. Since you mentioned using LazyVim, ensure your lspconfig.lua is correctly set up.

2. Use tsserver Directly
Ensure that typescript-language-server is using the correct tsserver. You can specify the path to tsserver explicitly in your LSP configuration:
```
require('lspconfig').tsserver.setup {
  cmd = { "typescript-language-server", "--stdio" },
  on_attach = function(client, bufnr)
    -- custom on_attach logic
  end,
  settings = {
    -- settings here
  },
}

```

3. Update Plugins and Language Server
Ensure all your plugins and the language server itself are up to date. Sometimes issues are resolved in newer versions. You can update your npm packages globally:
```
npm update -g typescript typescript-language-server

```

4. Increase Memory Limit
Sometimes increasing the memory limit for tsserver can help with performance issues. You can try increasing the Node.js memory limit:
```
export NODE_OPTIONS=--max_old_space_size=4096

```
